app/models/github_status.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+class GithubStatus
+  Status = Struct.new(:context, :latest_status) do
+    def state
+      latest_status.state
+    end
+
+    def description
+      latest_status.description
+    end
+
+    def url
+      latest_status.target_url
+    end
+
+    def success?
+      state == "success"
+    end
+
+    def failure?
+      state == "failure"
+    end
+
+    def pending?
+      state == "pending"
+    end
+  end
+
+  def initialize(repo, ref, github: GITHUB)
+    @github = github
+    @repo = repo
+    @ref = ref
+    @state = nil
+  end
+
+  def state
+    status_response ? status_response.state : "missing"
```

Comment:
here no status_response means it's missing ... and then below it is from empty statuses .. seems kinda confusing

Reply:
:+1:

Reply:
Changed the code so that we don't try to second-guess the API response's state.